--- ./src/ruby_supportlib/phusion_passenger/native_support.rb
+++ ./src/ruby_supportlib/phusion_passenger/native_support.rb
@@ -49,7 +49,8 @@ def start
         return false
       elsif try_load
         return true
-      elsif compile_and_load || download_binary_and_load
+      # This will not work in a read-only Nix environment
+      elsif ENV['PASSENGER_ALLOW_COMPILE_NATIVE_SUPPORT'] == '1' && (compile_and_load || download_binary_and_load)
         STDERR.puts " [#{library_name}] successfully loaded."
         return true
       else
